File: core/core-web/src/test/java/ch/difty/scipamato/core/web/CorePageParametersTest.java
package ch.difty.scipamato.core.web;

import static ch.difty.scipamato.core.web.CorePageParameters.*;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class CorePageParametersTest {

    @Test
    void values() {
        assertThat(CorePageParameters.values()).containsExactly(SEARCH_ORDER_ID, SHOW_EXCLUDED, USER_ID, MODE,
            TAB_INDEX);
    }

    @Test
    void assertRoleNames() {
        assertThat(SEARCH_ORDER_ID.getName()).isEqualTo("searchOrderId");
        assertThat(SHOW_EXCLUDED.getName()).isEqualTo("showExcluded");
        assertThat(USER_ID.getName()).isEqualTo("userId");
        assertThat(MODE.getName()).isEqualTo("mode");
        assertThat(TAB_INDEX.getName()).isEqualTo("tabIndex");
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/CorePageParametersTest.kt
package ch.difty.scipamato.core.web

import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldContainSame
import org.junit.jupiter.api.Test

internal class CorePageParametersTest {

    @Test
    fun values() {
        CorePageParameters.values() shouldContainSame listOf(
            CorePageParameters.SEARCH_ORDER_ID,
            CorePageParameters.SHOW_EXCLUDED,
            CorePageParameters.USER_ID,
            CorePageParameters.MODE,
            CorePageParameters.TAB_INDEX
        )
    }

    @Test
    fun assertRoleNames() {
        CorePageParameters.SEARCH_ORDER_ID.getName() shouldBeEqualTo "searchOrderId"
        CorePageParameters.SHOW_EXCLUDED.getName() shouldBeEqualTo "showExcluded"
        CorePageParameters.USER_ID.getName() shouldBeEqualTo "userId"
        CorePageParameters.MODE.getName() shouldBeEqualTo "mode"
        CorePageParameters.TAB_INDEX.getName() shouldBeEqualTo "tabIndex"
    }
}
